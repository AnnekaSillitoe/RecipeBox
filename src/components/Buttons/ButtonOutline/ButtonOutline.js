import React from "react";
import PropTypes from "prop-types";

const ButtonOutline = ({ buttonClasses="", onClick, buttonText="" }) => (
  <button
    className={"btn btn-outline-dark " + buttonClasses}
    onClick={onClick}
  >
    {buttonText}
  </button>
);

ButtonOutline.propTypes = {
  onClick: PropTypes.func.isRequired,
  buttonText: PropTypes.string.isRequired,
  buttonClasses: PropTypes.string
};

export default ButtonOutline;