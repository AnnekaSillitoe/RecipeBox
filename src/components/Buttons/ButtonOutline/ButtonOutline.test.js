import React from "react";
import { shallow } from "enzyme";
import ButtonOutline from "../ButtonOutline/ButtonOutline";

const mockEvent = jest.fn();
const props = {
  onClick: mockEvent,
  buttonText: "Recipes"
};
test("ButtonPrimary renders correctly", () => {
  const button = shallow(<ButtonOutline {...props} />);

  expect(button).toMatchSnapshot();
});

test("When clicking the button, it triggers the onClick", () => {
  const button = shallow(<ButtonOutline {...props} />);

  button.simulate("click");

  expect(mockEvent.mock.calls.length).toEqual(1);
});

test("Renders text passed through buttonText prop", () => {
  const button = shallow(<ButtonOutline {...props} />);
  expect(button.text()).toBe("Recipes");
});

test("handles classNames passed in through props", () => {
  const button = shallow(<ButtonOutline {...props} buttonClasses="blue" />);
  expect(button.props().className).toEqual(expect.stringContaining("blue"));
});
