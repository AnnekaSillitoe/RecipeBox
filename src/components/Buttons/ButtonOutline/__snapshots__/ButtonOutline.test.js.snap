// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPrimary renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonOutline
    buttonText="Recipes"
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Recipes",
      "className": "btn btn-outline-dark ",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Recipes",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Recipes",
        "className": "btn btn-outline-dark ",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Recipes",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
